### PR TITLE
bazel: update rules_foreign_cc

### PR DESCRIPTION
**Pull request by keith**
This update is required for this commit
https://github.com/bazelbuild/rules_foreign_cc/commit/c41020e4655fc9adb4bad7e9c04a3f2cf2cc9250
in order to support Apple Silicon. Since our last update there have been
some breaking API changes. I followed these instructions to migrate:

https://github.com/bazelbuild/rules_foreign_cc/releases/tag/0.3.0
https://github.com/bazelbuild/rules_foreign_cc/releases/tag/0.4.0

This was reverted once because of a regression fixed by https://github.com/bazelbuild/rules_foreign_cc/commit/da8952e27c46659866b455de58f5eb3a86185507

It was reverted again because we saw a failure on envoy-mobile, but that issue turned out to be a configuration issue there, not the fault of this change. https://github.com/bazelbuild/rules_foreign_cc/pull/789 provides a better error for that case in the future, but isn't required

This reverts commit 7760bc0e02ad32a6016c91ba53486f6bf87f004e.

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

**Comment by keith** [comment]
Ok this is ready now, updated the description with the reasoning

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -664,12 +664,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Rules for using foreign build systems in Bazel",
         project_desc = "Rules for using foreign build systems in Bazel",
         project_url = "https://github.com/bazelbuild/rules_foreign_cc",
-        version = "d54c78ab86b40770ee19f0949db9d74a831ab9f0",
-        sha256 = "e7446144277c9578141821fc91c55a61df7ae01bda890902f7286f5fd2f6ae46",
+        version = "da8952e27c46659866b455de58f5eb3a86185507",
+        sha256 = "5d1e34fc0f4ab4af0e595ee4d9a03f4127701d95c1d521296bf892b8aff903a2",
```
Comment (moderation) [inline bazel/repository_locations.bzl:668]: I know this has passed CI but do we want to bump to the latest commit? https://github.com/bazelbuild/rules_foreign_cc/commit/6c0c2af3d599f4c23117a5e65e811ebab75bb151

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -664,12 +664,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         strip_prefix = "rules_foreign_cc-{version}",
         urls = ["https://github.com/bazelbuild/rules_foreign_cc/archive/{version}.tar.gz"],
-        release_date = "2020-10-26",
-        use_category = ["build"],
+        release_date = "2021-08-21",
```
Comment (moderation) [inline bazel/repository_locations.bzl:671]: Latest commit linked to above is `2021-09-22`

**Review by moderation — COMMENTED** [review]
@keith do you want to bump to the latest commit or do that in follow up?

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (keith) [reply bazel/repository_locations.bzl:668]: done

**Comment by htuch** [comment]
@keith there are so many reverts and reverts-of-reverts that my brain (and probably any other newcomer) is unable to process. Can you give a summary for posterity of how we got into the situation that we had so many reverts? What's the future for this?

**Comment by keith** [comment]
I've updated the description with the original justification, as well as the revert history. The red CI appears to be unrelated

**Review by htuch — APPROVED** [review]
LGTM, thanks! I think it would be good to have better early warning from envoy-mobile to address this as a post-commit action item.

Is it possible to have envoy-mobile as a non-blocking CI check?

**Comment by keith** [comment]
totally, I submitted https://github.com/envoyproxy/envoy-mobile/issues/1831 for discussion

**Pull request merged**